config/src/test/java/org/springframework/security/config/web/server/ServerHttpSecurityTests.java

```diff
@@ -279,6 +283,33 @@ public void basicWithCustomRealmName() {
 		assertThat(result.getResponseCookies().getFirst("SESSION")).isNull();
 	}
 
```

Comment:
I'd like one more test that verifies that everything works with just `http.x509()` (the `principalExtractor` and `authenticationManager` should be optional). If any code changes are necessary for the test to pass make those too.

Reply:
That was indeed a very good point. Making default configuration to work forced me to think about the whole solution a little bit more. So to get feature parity with Servlet x509 I've added `ReactivePreAuthenticatedAuthenticationManager` which fetches user details using `ReactiveUserDetailsService` and checks account status. Account status check is delegated to  `AccountStatusUserDetailsChecker` using `ReactiveDelegatingUserDetailsChecker` to bridge reactive world with blocking. 

I've also added `ReactiveUserDetailsChecker`interface so developers can implement a proper non-blocking account check if they want to.